### PR TITLE
[CI] [GHA] Introduce GHA Linux Android ARM64 Pipeline

### PR DESCRIPTION
**Pull request by akashchi**
### Details:
This PR introduces the Linux Android ARM64 pipeline in the GitHub Actions environment. It is mirroring the [Azure Linux Android ARM64 pipeline](https://github.com/openvinotoolkit/openvino/blob/master/.ci/azure/android_arm64.yml).

### Tickets:
 - 117540

### CODE REVIEW

**Review by ilya-lavrenov — APPROVED** [review]

File: .github/workflows/linux_android_arm64.yml [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+#  push:
+#    paths-ignore:
+#      - '**/docs/**'
+#      - 'docs/**'
+#      - '**/**.md'
+#      - '**.md'
+#      - '**/layer_tests_summary/**'
+#      - '**/conformance/**'
+#    branches:
+#      - master
```
Comment (ilya-lavrenov) [inline .github/workflows/linux_android_arm64.yml:24]: should we remove commented code?

File: .github/workflows/linux_android_arm64.yml [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+
+concurrency:
+  group: ${{ github.head_ref || github.run_id }}-linux-android-arm64
+  cancel-in-progress: true
+
+jobs:
+  Build:
+    # TODO: remove. Temporary measure to prevent the workflow from scheduling on forks.
```
Comment (ilya-lavrenov) [inline .github/workflows/linux_android_arm64.yml:32]: what are the long term solution for future?

File: .github/workflows/linux_android_arm64.yml [hunk 3]
```diff
@@ -0,0 +1,178 @@
...
+        with:
+          repository: 'microsoft/vcpkg'
+          path: 'vcpkg'
+          fetch-depth: '0'
+          submodules: 'recursive'
+
+      - name: Create Directories
+        run: |
+          mkdir -p ${{ env.BUILD_DIR }}
+          mkdir -p ${{ env.INSTALL_DIR }}
```
Comment (ilya-lavrenov) [inline .github/workflows/linux_android_arm64.yml:71]: can we eliminate this step ? (I hope the directories are automatically created) or at least merge with other steps.

Ideally, we need steps which perform some interesting steps of pipeline (e.g. clone, install dependencies, build, test, package) and avoid dummy steps like create dir, print some values or simiar

File: .github/workflows/linux_android_arm64.yml [hunk 4]
```diff
@@ -0,0 +1,178 @@
...
+          key: ${{ github.job }}-linux-android-arm64
+          restore-keys: |
+            ${{ github.job }}-linux-android-arm64
+
+      - name: Get tools versions
+        run: |
+          ninja --version
+          ccache --version
+          python3 --version
+          cmake --version
```
Comment (ilya-lavrenov) [inline .github/workflows/linux_android_arm64.yml:123]: let's remove

File: .github/workflows/linux_android_arm64.yml [hunk 5]
```diff
@@ -0,0 +1,178 @@
...
+            -DENABLE_SYSTEM_PUGIXML=ON \
+            -DENABLE_SYSTEM_SNAPPY=ON \
+            -DENABLE_SYSTEM_TBB=ON \
+            -DENABLE_SYSTEM_FLATBUFFERS=ON \
+            -DENABLE_INTEL_GPU=ON \
+            -DENABLE_TESTS=ON \
+            -DCMAKE_CXX_COMPILER_LAUNCHER=ccache \
+            -DCMAKE_C_COMPILER_LAUNCHER=ccache \
+            -DCMAKE_CXX_LINKER_LAUNCHER=ccache \
+            -DCMAKE_C_LINKER_LAUNCHER=ccache \
```
Comment (ilya-lavrenov) [inline .github/workflows/linux_android_arm64.yml:164]: let's remove LINKER cache in every pipeline to minimize cache size (also, it's low probable that binaries are reused as is from cache)

File: .github/workflows/linux_android_arm64.yml (hunk 2 above)
Comment (akashchi) [reply .github/workflows/linux_android_arm64.yml:32]: Once we have a sufficient # of runners/allowed parallel pipelines, these workflows will be not on the `schedule` trigger, but on a `pull_request` trigger to validate PRs. 
Now, they are enabled only on a nightly basis to check that they run correctly.

**Pull request merged**